File: library_frontend/lib/Models-Providers/book.dart
class Book {
  final int id;
  final String title;
  final String author;
  final int pageCount;
  final String genre;
  final String description;
  final double rating;
  final bool isAvailable;
  Book({required this.id, required this.title, required this.author, required this.isAvailable, required this.pageCount, required this.description, required this.genre, required this.rating});

  factory Book.fromJson(Map<String, dynamic> j) => Book(
        id: j['id'],
        title: j['title'],
        author: j['author'],
        isAvailable: j['isAvailable'] ?? true,
        pageCount: j['pageCount'] ?? 0,
        description: j['description'] ?? '',
        genre: j['genre'] ?? '',
        rating: (j['rating'] != null)
            ? (j['rating'] as num).toDouble()
            : 0.0,
      );
}

File: LibraryApp/Models/Book.cs
namespace LibraryApp.Models
{
    public class Book
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Genre { get; set; }
        public int? PageCount { get; set; }
        public string Description { get; set; }
        public double? Rating { get; set; }
        public string Author { get; set; }
        public bool IsAvailable { get; set; }
        public bool IsArchived { get; set; } = false;
        public ICollection<BorrowRecord> BorrowRecords { get; set; }
    }
}
